packages/strapi-design-system/src/Link/Link.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { TableLabel } from '../Text';
+import { Box } from '../Box';
+
+const LinkWrapper = styled.a`
+  text-transform: uppercase;
+  text-decoration: none;
+  pointer-events: ${({ disabled }) => (disabled ? 'none' : undefined)};
+
+  svg path {
+    fill: ${({ disabled, theme }) => (disabled ? theme.colors.neutral600 : theme.colors.primary600)};
+  }
+
+  svg {
+    font-size: ${10 / 16}rem;
+  }
+
+  &:hover,
+  &:active {
+    color: ${({ theme }) => theme.colors.primary800};
+  }
+`;
+
+// TODO: make sure to use the link from the router library chosen
+export const Link = ({ href, to, children, disabled, leftIcon, rightIcon, ...props }) => {
+  const linkHref = disabled ? '#' : href || to;
+  const target = href ? '_blank' : undefined;
+  const rel = href ? 'noreferrer noopener' : undefined;
+
+  return (
+    <LinkWrapper target={target} rel={rel} href={linkHref} disabled={disabled} {...props}>
+      <TableLabel textColor={disabled ? 'neutral600' : 'primary600'} as="span">
+        {leftIcon && (
+          <Box as="span" aria-hidden={true} paddingRight={2}>
+            {leftIcon}
+          </Box>
+        )}
+        {children}
+        {rightIcon && (
+          <Box as="span" aria-hidden={true} paddingLeft={2}>
+            {rightIcon}
+          </Box>
+        )}
+      </TableLabel>
+    </LinkWrapper>
+  );
+};
+
+Link.displayName = Link;
+
+Link.defaultProps = {
+  href: undefined,
+  to: undefined,
+  disabled: false,
+};
+
+Link.propTypes = {
+  children: PropTypes.node.isRequired,
+  disabled: PropTypes.bool,
+  href: PropTypes.string,
+  leftIcon: PropTypes.element,
+  rightIcon: PropTypes.element,
+  to: PropTypes.string,
```

Comment:
I think we should make the href or the to props required when the link is not disabled, WDYT of the following?

```
Link.propTypes = {
  href: (props, propName, componentName) => {
    if (!props.disabled && !props.to && !props.href) {
      return new Error("href must be defined");
    }
  },
  to: (props) => {
    if (!props.disabled && !props.href && !props.to) {
      return new Error("to must be defined");
    }
  }
};
```

Reply:
I can do that :)